gmso/formats/top.py

```diff
@@ -151,11 +161,10 @@ def write_top(top, filename, top_vars=None):
                         )
                     )
 
-            # Special treatment for water, may ned to consider a better way to tag rigid water
+            # Special treatment for water, may ned to consider a better way to tag rigid SETTLES for 3-site water
+            # need a more general way to work for 4-site and 5-site models.
             # Built using this https://github.com/gromacs/gromacs/blob/main/share/top/oplsaa.ff/spce.itp as reference
-            if "water" in tag.lower() and all(
-                site.molecule.isrigid for site in unique_molecules[tag]["sites"]
-            ):
+            if settles_tag and settles_tag.lower() == tag.lower():
```

Comment:
Before, we checked for the tag name, and `site.molecule.isrigid`. Do we not need to check `site.molecule.isrigid` anymore with these changes?

Reply:
Good question. Per @dyukovsm and my discussion, we think isrigid is different than SETTLES. Any molecule can be treated with isrigid. However, SETTLES is exclusive to just water. So we think the manual check for SETTLES can be explicitly for 3-site water, whereas we probably can hope to incorporate a more general treatment if a molecule has `isrigid()`, which would be entirely different from the SETTLES treatment.